src/common/config_parser.cpp

```diff
@@ -70,12 +70,13 @@ std::string const& ConfigParser::cmdLine() const {
 }
 
 ConfigParser::ConfigParser(cli::mode mode)
-  : cli_(config_,"Marian: Fast Neural Machine Translation in C++",
-         "General options", "", 40),
-    mode_(mode == cli::mode::server ? cli::mode::translation : mode) {
+    : cli_(config_, "Marian: Fast Neural Machine Translation in C++", "General options", "", 40),
+      mode_(mode == cli::mode::server
+                ? cli::mode::translation
+                : (mode == cli::mode::selfadaptiveServer ? cli::mode::selfadaptive : mode)) {
```

Comment:
I think this requires a comment in the code to explain what is being done here. Is it just to replace `cli::mode::selfadaptiveserver` to `cli::mode::selfadaptive` early on?

Reply:
Made the code more explicit and added comments